sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,237 @@
+        # code should be modified accordingly. At the time to consider the
+        # sample_prop feature, it is also a good use case to be considered.
+        self.transformer_.fit(y)
+        if self.check_inverse:
+            random_state = check_random_state(self.random_state)
+            n_subsample = min(10, y.shape[0])
+            subsample_idx = random_state.choice(range(y.shape[0]),
+                                                size=n_subsample,
+                                                replace=False)
+            if not np.allclose(
+                    y[subsample_idx],
+                    self.transformer_.inverse_transform(
+                        self.transformer_.transform(y[subsample_idx]))):
+                raise ValueError("The provided functions or transformer are"
+                                 " not strictly inverse of each other. If"
+                                 " you are sure you want to proceed regardless"
+                                 ", set 'check_inverse=False'")
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape (n_samples,) optional
+            Array of weights that are assigned to individual samples.
+            If not provided, then each sample is given unit weight.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y = check_X_y(X, y, multi_output=True, y_numeric=True)
+
+        # transformer are designed to modify X which is a 2d dimensional, we
+        # need to modify y accordingly.
+        if y.ndim == 1 and self.func is None:
+            y_2d = y.reshape(-1, 1)
+        else:
+            y_2d = y
+        self._fit_transformer(y_2d)
+
+        if self.regressor is None:
+            from ..linear_model import LinearRegression
+            self.regressor_ = LinearRegression()
+        else:
+            self.regressor_ = clone(self.regressor)
+
+        support_sample_weight = has_fit_parameter(self.regressor_,
+                                                  'sample_weight')
+        if not support_sample_weight and sample_weight is not None:
+            raise ValueError("The regressor {} does not support sample "
+                             "weight.".format(
+                                 self.regressor_.__class__.__name__))
+
+        # transform y and convert back to 1d array if needed
+        y_trans = self.transformer_.fit_transform(y_2d)
+        if y.ndim == 1 and self.func is None:
+            y_trans = y_trans.reshape(-1)
```

Comment:
Might want to check that this is going to be the right number of elements. I suppose you could conceive of a transformer which changed single-output to multi-output (though more likely the opposite)?